spec/PluginClientSpec.php

```diff
@@ -69,4 +69,19 @@ function it_throws_loop_exception(HttpClient $client, RequestInterface $request)
 
         $this->shouldThrow('Http\Client\Plugin\Exception\LoopException')->duringSendRequest($request);
     }
+
+    function it_adds_a_plugin(Plugin $plugin)
+    {
+        $this->addPlugin($plugin);
+
+        $this->getPlugins()->shouldReturn([$plugin]);
+    }
+
+    function it_sets_new_plugins()
+    {
+        $plugins = [new LoopPlugin];
```

Comment:
This isn't really. A mock would be better here as well, like in case of the add example.